hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -239,20 +247,20 @@ createOutputAtAddress networkId backend atAddress datum val = do
         Just u -> pure u
 
 -- | Try to submit tx and retry when some caught exception/s take place.
-retryOnExceptions :: (MonadCatch m, MonadDelay m) => Tracer m FaucetLog -> m a -> m a
-retryOnExceptions tracer action =
+retryOnExceptions :: (MonadCatch m, MonadDelay m, ChainBackend backend) => Tracer m FaucetLog -> backend -> m a -> m a
+retryOnExceptions tracer backend action =
   action
     `catches` [ Handler $ \(_ :: SubmitTransactionException) -> do
-                  threadDelay 1
-                  retryOnExceptions tracer action
+                  delayBF backend
+                  retryOnExceptions tracer backend action
               , Handler $ \(ex :: IOException) -> do
                   unless (isResourceExhausted ex) $
                     throwIO ex
                   traceWith tracer $
                     TraceResourceExhaustedHandled $
                       "Expected exception raised from seedFromFaucet: " <> show ex
-                  threadDelay 1
-                  retryOnExceptions tracer action
+                  delayBF backend
+                  retryOnExceptions tracer backend action
```

Comment:
if retryOnExceptions now takes the backend, perhaps we want to to call the delay in there.

wdyt?

Reply:
We are literary in `retryOnExceptions` function here :)